governance/api.go

```diff
@@ -62,22 +63,33 @@ var (
 	errInvalidUpperBound      = errors.New("upperboundbasefee cannot be set lower than lowerboundbasefee")
 )
 
-// TODO-Klaytn-Governance: Refine this API and consider the gas price of txpool
+// GasPriceAt returns the baseFeePerGas of the given block in peb.
 func (api *GovernanceKlayAPI) GasPriceAt(num *rpc.BlockNumber) (*hexutil.Big, error) {
-	if num == nil || *num == rpc.LatestBlockNumber || *num == rpc.PendingBlockNumber {
-		ret := api.governance.UnitPrice()
-		return (*hexutil.Big)(big.NewInt(0).SetUint64(ret)), nil
+	if num == nil || *num == rpc.LatestBlockNumber {
+		header := api.chain.CurrentHeader()
+		if header.BaseFee == nil {
+			return (*hexutil.Big)(new(big.Int).SetUint64(api.governance.UnitPrice())), nil
+		}
+		return (*hexutil.Big)(header.BaseFee), nil
+	} else if *num == rpc.PendingBlockNumber {
+		txpool := api.governance.GetTxPool()
+		return (*hexutil.Big)(txpool.GasPrice()), nil
 	} else {
-		blockNum := num.Int64()
+		blockNum := num.Uint64()
 
-		if blockNum > api.chain.CurrentHeader().Number.Int64() {
+		if blockNum > api.chain.CurrentHeader().Number.Uint64() {
 			return nil, errUnknownBlock
 		}
 
+		header := api.chain.GetHeaderByNumber(blockNum)
```

Comment:
Don't we need to handle `header == nil` case here?

Reply:
Good point.
Added the condition. Thanks!